src/cryptography/hazmat/primitives/serialization/pkcs7.py

```diff
@@ -263,6 +263,81 @@ def encrypt(
         return rust_pkcs7.encrypt_and_serialize(self, encoding, options)
 
 
+def pkcs7_decrypt_der(
```

Comment:
I think we should move all these functions to rust and export them here.

Reply:
Which also means, passing the smime-related function in Python as private, and calling it on Rust, I guess?

Reply:
If you look at `pkcs7.rs` and search for `SMIME_ENVELOPED_ENCODE` you can see how we call into the S/MIME Python from Rust.

Reply:
Done! One last question: should I also move the logic of the new _check_pkcs7_decrypt_arguments function to rust, or just leave it as such and call it the same way I'm calling SMIME_ENVELOPED_DECODE?

Reply:
Yes, let's move that as well -- it actually will probably become a fair bit easier, as many things are automatically handled by giving parameters types.

Reply:
Should be done. I'd love a quick review on the `check_decrypt_parameters` function, it is probably more intricate than it should be!